platform-api/che-core-api-auth/src/main/java/org/eclipse/che/security/oauth/OAuthAuthenticationService.java

```diff
@@ -236,6 +239,20 @@ public Response invalidate() throws IOException {
                                                                            null)).type(MediaType.TEXT_PLAIN).build();
     }
 
+    @GET
+    @Produces({MediaType.APPLICATION_JSON})
+    @RolesAllowed({"user", "temp_user"})
+    @Path("token")
+    public OAuthToken token(@QueryParam("oauth_provider") String oauth_provider, @Context SecurityContext security)
+            throws Exception {
```

Comment:
Rename <code>oauth_provider</code> to <code>oauthProvider</code> (actually i am working on codestyle guide - it covers it as well). 
What if <code>oauthProvider</code> is <code>null</code>,  it must be documented!
Why do method throws <code>Exception</code>? It is better to use separate exceptions in this case such as <code>ConflictException</code> or <code>NotFoundException</code> and document it.